### ISSUE TITLE
BackgroundIndex need reindex when compile command changed

### ISSUE THREAD

**Issue by lh123** [clangd/clangd#199]
BackgroundIndex need reindex when compile command changed.

I think this problem may be caused by the function `shardIsStale`, which does not check if the compile command has changed.

**Comment by kadircet** [comment]
Yes, this was a performance/correctness trade-off we made. Maybe we can add a flag for this to trigger rebuilds whenever command line flags differs. WDYT @sam-mccall ?
